viewer/viewer-awt/src/main/java/org/icepdf/ri/util/ViewerPropertiesManager.java

```diff
@@ -278,32 +278,64 @@
     // stored state of last used public/private annotation flag.
     public static final String PROPERTY_ANNOTATION_LAST_USED_PUBLIC_FLAG = "application.viewer.annotation.public.flag";
 
-    private static PropertiesManager propertiesManager;
+    private static ViewerPropertiesManager propertiesManager;
 
     // static store of properties which are persisted to backing store.
-    private static Preferences preferences = Preferences.userNodeForPackage(PropertiesManager.class);
+    private static final Preferences preferences = Preferences.userNodeForPackage(ViewerPropertiesManager.class);
     // local properties, that aren't persisted and can override properties in the store if root accessor are used.
     private static Properties localProperties;
     // default properties file included int the viewer jar
     private static Properties defaultProps;
 
-    private PropertiesManager() {
+    private ViewerPropertiesManager() {
     }
 
     /**
      * Gets singleton instance of the the Properties manager instance.
      *
      * @return singleton instance.
      */
-    public static PropertiesManager getInstance() {
+    public static ViewerPropertiesManager getInstance() {
         if (propertiesManager == null) {
-            propertiesManager = new PropertiesManager();
-            // load default properties from viewer jar and assigned to defaultProps.
-            setupDefaultProperties();
+            propertiesManager = new ViewerPropertiesManager();
+            try {
+                if (preferences.keys().length == 0) {
+                    // load default properties from viewer jar and assigned to defaultProps.
+                    setupDefaultProperties();
+                    saveLocalProperties();
+                } else {
+                    updatePropertiesWithPreferences();
+                }
+            } catch (BackingStoreException e) {
+                logger.severe("Couldn't access backing store");
+                setupDefaultProperties();
+            }
         }
         return propertiesManager;
     }
 
+    /**
+     * Prints all the preferences and properties. Can be used to see discrepancies.
+     */
+    public static void printAllProperties() {
+        try {
+            Arrays.stream(preferences.keys()).forEach(key -> {
+                System.out.println(key + " -> " + preferences.get(key, ""));
```

Comment:
I'm on the fence here as I generally don't like to see system.out but you likely have this for debugging. Perhaps update the javadoc to explicitly mention prints to system.out?

Reply:
Oh right, it's mainly a debugging method, I could also remove it. For now I've changed the print to logger.fine